docs/src/modules/zend.service-manager.intro.rst

```diff
@@ -17,144 +22,179 @@ Following is the ``Zend\ServiceManager\ServiceLocatorInterface`` API:
        public function get($name);
        public function has($name);
    }
-	
+
 - ``has($name)``, tests whether the ``ServiceManager`` has a named service;
 
 - ``get($name)``, retrieves a service by the given name.
 
+.. _zend.service-manager.api:
+
+Additional API
+--------------
+
 A ``Zend\ServiceManager\ServiceManager`` is an implementation of the ``ServiceLocatorInterface``.
 In addition to the above described methods, the ``ServiceManager`` provides additional API:
 
-- **Service registration**. ``ServiceManager::setService`` allows you to register an object as a service:
+.. _zend.service-manager.api.service-registration:
 
-  .. code-block:: php
-     :linenos:
+Service registration
+^^^^^^^^^^^^^^^^^^^^
 
-     $serviceManager->setService('my-foo', new stdClass());
-     $serviceManager->setService('my-settings', array('password' => 'super-secret'));
+``ServiceManager::setService`` allows you to register an object as a service:
 
-     var_dump($serviceManager->get('my-foo')); // an instance of stdClass
-     var_dump($serviceManager->get('my-settings')); // array('password' => 'super-secret')
+.. code-block:: php
+   :linenos:
 
-- **Lazy-loaded service objects**. ``ServiceManager::setInvokableClass`` allows you to tell the
-  ``ServiceManager`` what class to instantiate when a particular service is requested:
+    $serviceManager->setService('my-foo', new stdClass());
+    $serviceManager->setService('my-settings', array('password' => 'super-secret'));
 
-  .. code-block:: php
-     :linenos:
+    var_dump($serviceManager->get('my-foo')); // an instance of stdClass
+    var_dump($serviceManager->get('my-settings')); // array('password' => 'super-secret')
 
-     $serviceManager->setInvokableClass('foo-service-name', 'Fully\Qualified\Classname');
+.. _zend.service-manager.api.lazy-loaded-objects:
 
-     var_dump($serviceManager->get('foo-service-name')); // an instance of Fully\Qualified\Classname
+Lazy-loaded service objects
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-- **Service factories**. Instead of an actual object instance or a class name, you can tell the
-  ``ServiceManager`` to invoke a provided factory in order to get the object instance. Factories
-  may be either a PHP `callback`_, an object implementing ``Zend\ServiceManager\FactoryInterface``,
-  or the name of a class implementing that interface:
+``ServiceManager::setInvokableClass`` allows you to tell the ``ServiceManager`` what class to instantiate when a
+particular service is requested:
 
-  .. code-block:: php
-     :linenos:
+.. code-block:: php
+   :linenos:
 
-     use Zend\ServiceManager\FactoryInterface;
-     use Zend\ServiceManager\ServiceLocatorInterface;
+   $serviceManager->setInvokableClass('foo-service-name', 'Fully\Qualified\Classname');
 
-     class MyFactory implements FactoryInterface
-     {
-         public function createService(ServiceLocatorInterface $serviceLocator)
-         {
-             return new \stdClass();
-         }
-     }
```

Comment:
You remove the example here, but then do not provide a full example later. For many, this is the first and/or only example they find in the manual. I think we need to keep it, albeit move it under the new section heading.